### PR TITLE
cylc gui: Use execution time limit as default mean_elapsed_time

### PR DESCRIPTION
**Pull request by arjclark**
This change makes it so that for the first run of a task the mean_elapsed_time defaults to using the `execution_time_limit` as a sensible default value if available. This makes to progress bar more informative when you are on the first run of a task and have a *sensible* execution time limit set for it rather than it being full from the start for a 2 hour job.

### CODE REVIEW

**Comment by arjclark** [comment]
@hjoliver @matthewrmshin (as you're looking at it now anyway) please review

File: lib/cylc/task_proxy.py [hunk 1]
```diff
@@ -1508,9 +1508,13 @@ def get_state_summary(self):
...
-        self.summary['mean_elapsed_time'] = (
-            float(sum(self.tdef.elapsed_times)) /
-            max(len(self.tdef.elapsed_times), 1))
+        count = len(self.tdef.elapsed_times)
+        if count == 0:
+            self.summary['mean_elapsed_time'] = \
+                max(self.summary['execution_time_limit'], 1)
+        else:
+            self.summary['mean_elapsed_time'] = (
+                float(sum(self.tdef.elapsed_times)) / count)
```
Comment (matthewrmshin) [inline lib/cylc/task_proxy.py]: You may want to reverse the block:

```python
if count:
    # calculate mean
elif self.summary['execution_time_limit']:
    # set to execution time limit
else:
    # zero?
```

Comment (arjclark) [inline lib/cylc/task_proxy.py]: I can have this default to 0 if that's preferable? I've used 1 here as a dummy value as it makes sense (to me) that if something runs its runtime is inherently a non-zero value. @hjoliver - any preference on a default before go change this?

Comment (hjoliver) [inline lib/cylc/task_proxy.py]: I like the general idea but I'm not sure I understand the 0 vs 1 value - `self.summary['execution_time_limit']` is a duration like `PT10M` here isn't it, so the `max` computation with 0 or 1 doesn't make sense?  And anyway, wouldn't it be best to show no progress bar in the first run of a task if no execution time limit is set?

Comment (arjclark) [inline lib/cylc/task_proxy.py]: The 0/1 is the existing behaviour just arguing semantics. I'll see what I can do about not rendering a progress bar in the absence of any mean execution time limit.

**Comment by arjclark** [comment]
@matthewrmshin - feedback addressed and logic reversed.

Behaviour was replicating what was already there in the absence of any time data to estimate a mean from. Behaviour now is, in the absence of previous runs and any best guess in the form of an execution time limit to just return None. This is different from before but is probably the sensible thing to do.

suite.rc you can play round with for seeing what behaviour is:
```
[scheduling]
initial cycle point = 20100101T0000Z
[[dependencies]]

[[[R//PT6H]]]
    graph = foo[-PT6H] => foo

[runtime]
[[foo]]
    script = sleep 10
    [[[job]]]
    execution time limit = PT60S # comment/uncomment as desired to see effect on 1st instance
```

**Pull request merged**